From: Michael Froman <mfroman@mozilla.com>
Date: Tue, 13 May 2025 22:38:00 +0000
Subject: Bug 1966238 - Cherry-pick upstream libwebrtc commit 3039b61768
 r?pehrsons!

Upstream commit: https://webrtc.googlesource.com/src/+/3039b617687c2bb2e7ff8adf2500f44140667dc3
       In ScreenCapturerSck don't restart on SelectSource if display didn't change

       Bug: webrtc:367915807
       Change-Id: I637b7f46a766fe9120f3f30c15f3903f5aec1d3d
       Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/365087
       Reviewed-by: Alexander Cooper <alcooper@chromium.org>
       Commit-Queue: Andreas Pehrson <apehrson@mozilla.com>
       Cr-Commit-Position: refs/heads/main@{#44210}

Differential Revision: https://phabricator.services.mozilla.com/D249156
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/d463d67dfde5b7db6c8df4804b48f9c57137378d
---
 modules/desktop_capture/mac/screen_capturer_sck.mm | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/modules/desktop_capture/mac/screen_capturer_sck.mm b/modules/desktop_capture/mac/screen_capturer_sck.mm
index 9379940504..479b5ba666 100644
--- a/modules/desktop_capture/mac/screen_capturer_sck.mm
+++ b/modules/desktop_capture/mac/screen_capturer_sck.mm
@@ -203,6 +203,9 @@ bool ScreenCapturerSck::SelectSource(SourceId id) {
   bool stream_started = false;
   {
     MutexLock lock(&lock_);
+    if (current_display_ == id) {
+      return true;
+    }
     current_display_ = id;
 
     if (stream_) {
